src/main/java/org/gridsuite/study/server/StudyController.java

```diff
@@ -102,308 +100,282 @@ public StudyController(StudyService studyService) {
         return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(studyService.studyExists(studyName, userId));
     }
 
-    @DeleteMapping(value = "/{userId}/studies/{studyName}")
+    @DeleteMapping(value = "/studies/{studyUuid}")
     @ApiOperation(value = "delete the study")
     @ApiResponse(code = 200, message = "Study deleted")
-    public ResponseEntity<Mono<Void>> deleteStudy(@PathVariable("studyName") String studyName,
-                                                  @PathVariable("userId") String userId,
+    public ResponseEntity<Mono<Void>> deleteStudy(@PathVariable("studyUuid") UUID studyUuid,
                                                   @RequestHeader("userId") String headerUserId) {
-        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(studyService.assertUserAllowed(userId, headerUserId)
-                .doOnSuccess(s -> studyService.deleteStudyIfNotCreationInProgress(studyName, userId).subscribe()));
+        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(
+                studyService.deleteStudyIfNotCreationInProgress(studyUuid, headerUserId));
```

Comment:
Why make this synchronous ? why was it asynchronous before ?

Reply:
discussed together, we need to get the user id from the database anyway now, so in the same transaction delete the study